tests/integration/utils.py

```diff
@@ -121,10 +123,25 @@ def calculate_files_sha256sum_in_dir(root_dir: str) -> Dict:
         for file_name in files:
             rel_dir = os.path.relpath(dir_, root_dir)
             rel_file = os.path.join(rel_dir, file_name)
-            files_checksums[rel_file] = calculate_sha256sum(os.path.join(root_dir, rel_file))
+            if "-external-gitcommit-" in file_name:
+                files_checksums[rel_file] = _get_git_commit_from_tarball(
+                    os.path.join(root_dir, rel_file)
+                )
+            else:
+                files_checksums[rel_file] = calculate_sha256sum(os.path.join(root_dir, rel_file))
     return files_checksums
 
 
+def _get_git_commit_from_tarball(tarball: str) -> str:
+    tarfile = TarFile.open(tarball, "r:gz")
+    extract_path = tarball.replace(".tar.gz", "")
+
+    safe_extract(tarfile, extract_path)
+    repo = Repo(path=f"{extract_path}/app")
+
+    return f"gitcommit:{repo.commit().hexsha}"
```

Comment:
Just like regular `open()`, the object returned by `tarfile.open()` must be closed (ideally by using it as a context manager)

This function also leaves the extracted content lying around on disk. What happens when you call this function twice with the same argument? Probably fails?